core/runtime/src/main/java/io/quarkus/runtime/configuration/ssl/ServerSslConfig.java

```diff
@@ -126,8 +127,9 @@ public SSLContext toSSLContext() throws GeneralSecurityException, IOException {
             if (keyItr.hasNext()) {
                 log.warnf("Ignoring extra content in key file \"%s\"", keyPath);
             }
+            //Entry password needs to match the keystore password
             keyStore.setEntry("default", new KeyStore.PrivateKeyEntry(privateKey, certList.toArray(new X509Certificate[0])),
-                    new KeyStore.PasswordProtection("password".toCharArray()));
+                    new KeyStore.PasswordProtection(keystorePassword.toCharArray()));
```

Comment:
Don't you also have to deal with it if a key store file is provided just below?

Right now, it's loaded like that: `keyStore.load(is, null);`?

Can we create a key store file with no password? If so, do we need to handle different defaults for the password?

Reply:
I was wondering the same thing initially, but the 
[load](https://docs.oracle.com/javase/8/docs/api/java/security/KeyStore.html#load-java.io.InputStream-char:A-) implies that it is not necessary to provide a password, only for integrity checks. Later on, the KeyManagerFactory actually takes care of loading the keystore (either the one provided or created) using a password.

I tried to be as unobtrusive with the code as possible, but I could try to break the method `toSSLContext` up in smaller bits, as to keep the flow more clear:
1. Figure out what config is provided 
2. Either load or create a key store
3. Build the SSL context

What I did to test it:
1. I generated a certificate using OpenSSL and provided the certificate and private key to Quarkus
2. I created a keystore and provided a keystore and the new password configuration item to Quarkus

What did I base both scenarios on? The code implies both options are mutually exclusive, since it will either try to _create_ a keystore with cert/key and otherwise try to load the provided keystore.

Both methods worked (otherwise I wouldn't submit it for a PR of course :-)), and I would like to add the above methods as automated tests, for example using a cURL command to verify that HTTP is actually working.

Reply:
@fndejong OK, looks good then. We need tests though: could you try to add some integration tests in the undertow/deployment module? There are already a couple of integration tests there. You can use REST assured to test things, no need for curl.

I'm also wondering if you would be interesting in writing some documentation there: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/rest-json-guide.adoc . A new section just before the conclusion would be nice. It's not the perfect place in the long term but for now, it will do.

Reply:
@gsmet : I'll look into both the tests and documentation. I hope I can give an update tomorrow.